opencti-platform/opencti-graphql/src/modules/ai/ai-domain.ts

```diff
@@ -294,3 +302,67 @@ export const convertFilesToStix = async (context: AuthContext, user: AuthUser, a
   const response = await queryAi(id, SYSTEM_PROMPT, prompt, user);
   return response;
 };
+
+export const filtersEntityIdsMapping = async (context: AuthContext, user: AuthUser, filters: FilterGroup) => {
+  // 01. fetch the filter keys corresponding to an id
+  const filterDefinitions = await generateFilterKeysSchema();
+  const stixCoreObjectsFilterDefinitions = filterDefinitions.find((f) => f.entity_type === ABSTRACT_STIX_CORE_OBJECT)?.filters_schema.map((f) => f.filterDefinition) ?? [];
+  const idsFilterKeys = stixCoreObjectsFilterDefinitions.filter((f) => f.type === 'id')
+    .map((f) => f.filterKey)
+    .concat([INSTANCE_REGARDING_OF]);
+  // 02. fetch the values to resolve in the filter group
+  const valuesIdsToResolve = extractFilterGroupValues(filters, idsFilterKeys);
+  // 03. resolve the values and deduce potential corresponding ids
+  const notResolvedValues: string[] = [];
+  const mapContent = await Promise.all(valuesIdsToResolve.map(async (value): Promise<[string, string | null]> => {
+    const stixCoreObjectsFilter = addFilter(undefined, 'entity_type', ['Stix-Core-Object']);
+    const result = await findAll(context, user, {
+      filters: stixCoreObjectsFilter,
+      search: value,
+      orderBy: '_score',
+      orderMode: 'desc',
+    });
+    const resultIds = result.edges.map((n) => n.node.id);
+    if (resultIds.length > 0) {
+      return [value, resultIds[0]];
+    }
+    notResolvedValues.push(value);
+    return [value, null];
+  }));
+  const valuesIdsMap = new Map(mapContent);
+  // 04. replace the values with their corresponding ids
+  return {
+    filters: filtersEntityIdsMappingResult(filters, idsFilterKeys, valuesIdsMap),
+    notResolvedValues
+  };
+};
+
+export const generateNLQresponse = async (context: AuthContext, user: AuthUser, args: MutationAiNlqArgs) => {
+  await checkEnterpriseEdition(context);
+  const { search } = args;
+  const promptValue = await NLQPromptTemplate.formatPromptValue({ text: search });
+
+  // 01. query the model
+  logApp.debug('[AI] Querying NLQ with prompt', { questionStart: search.substring(0, 100) });
+  const NLQ_TIMEOUT = 30 * 1000; // timeout: 30s
+  let rawResponse;
+  try {
+    rawResponse = await callWithTimeout(queryNLQAi(promptValue), NLQ_TIMEOUT);
+  } catch (e) {
+    throw UnknownError('The NLQ model takes too long to respond', { promptValue });
```

Comment:
It's not necessary a timeout, and we need 'e' in meta data to have more informations in logs.